### PR TITLE
created category card

### LINKED ISSUE #106: (SP: 1) Implement category item card
## Description
Implement category card so there will be icon, title, description (amount of offers for this category). Also, there should be a link so if a user clicks on the card a page should redirect to it.
## Design 

Image

left - hovered, right - normal
## Resources
Links to the documentations, libraries, guides 
## Acceptance Criteria
Can be copied from user story or give more detailed, we can use 
- [ ] the block should be responsive to hover
- [ ] when a user clicks on the card the should be redirected to "Subjects" page with selected category

### LINKED ISSUE #107: (SP: 2) [Categories] Implement list of categories block
**Blocked by** #106 
## Description
Implement list of categories block with amount of offers related to it
## Design 

Image

## Acceptance Criteria
- [ ] When user clicks on category card, she/he should be redirected to "subjects" page with list of all subjects related to that category
- [ ] "View more" button should render more cards 
- [ ] "View more" button should disapear if there are no more cards to display

### LINKED ISSUE #108: (SP: 1) [Categories] Add title with description and search input
## Description
Add title with description, implement input with searching functionality to search for a category by name
## Design 

Image

## Acceptance Criteria
- [ ] When a user clicks on the "Show all offers", he/she should be redirected to "find offers" page
- [ ] When a user clicks on the "Search" button or press enter, search should be performed

### PR DESCRIPTION
**Pull request by HalynaProlubshchykova**
For now, the page is static.
The placement of cards is adaptive for different screen sizes
For desktop
desktop
For tablet
tablet
For mobile
mobile

### CODE REVIEW

**Review by MrAK-dev — APPROVED** [review]

**Review by tufned — APPROVED** [review]

**Review by tufned — COMMENTED** [review]
it is better to maintain the same format of styles for the project: `.styles.js` instead of `.css`

like this:

```
<div style={styles.categories} />
```
in Categories.styles.js
``` 
export const styles = {
  categories: {
    display: 'grid',
    gridTemplateColumns: 'repeat(3, 1fr)',
    gap: '20px',
  }
}
```

Maybe you could change it in the future?
Anyway, I'm approving for now, so that we can show more finished stuff on demo)

**Review by grbvtsk — APPROVED** [review]

**Review by SachkovaYuliia — APPROVED** [review]

**Review by olehpona — APPROVED** [review]

**Review by vlad-khrychov — APPROVED** [review]

File: src/pages/categories/Categories.tsx [hunk 1]
```diff
@@ -1,7 +1,76 @@
...
+import CategoriesHeader from './CategoriesHeader'
 
-const Categories = () => {
-  return <PageWrapper>Categories</PageWrapper>
+const Categories: React.FC = () => {
+  const [visibleCount, setVisibleCount] = useState<number>(6)
+  const [filteredCategories, setFilteredCategories] = useState(categoriesData)
+
+  const handleViewMore = () => {
+    setVisibleCount((prevCount) => prevCount + 3)
```
Comment (vlad-khrychov) [inline src/pages/categories/Categories.tsx:13]: Save 3 and 6 into const, maybe we can change it for different screen sizes in future.

File: src/pages/categories/Categories.tsx [hunk 2]
```diff
@@ -1,7 +1,76 @@
...
+  return (
+    <PageWrapper>
+      <CategoriesHeader onSearch={handleSearch} />
+      {filteredCategories.length > 0 ? (
+        <>
+          <Grid container justifyContent='center' spacing={4}>
+            {filteredCategories
+              .slice(0, visibleCount)
+              .map((category, index) => (
+                <Grid item key={index} md={4} sm={6} xs={12}>
```
Comment (vlad-khrychov) [inline src/pages/categories/Categories.tsx:36]: How it looks if we show 9 items? Do we have empty space in the last row?

File: src/pages/categories/CategoriesHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+interface CategoriesHeaderProps {
+  onSearch: (searchQuery: string) => void
+}
+
+const CategoriesHeader: React.FC<CategoriesHeaderProps> = ({ onSearch }) => {
+  const [searchQuery, setSearchQuery] = useState('')
+  const navigate = useNavigate()
+
+  const handleShowOffers = () => {
+    navigate('/categories/subjects/find-offers')
```
Comment (vlad-khrychov) [inline src/pages/categories/CategoriesHeader.tsx:22]:
```suggestion
    navigate('/categories/${id}/subjects/find-offers')
```
And change routing for this

**Review by vlad-khrychov — APPROVED** [review]
All comments can be done in the next PR.

**Pull request merged**